lib/index.ts

```
@@ -0,0 +1,468 @@
+        new Promise(resolve => import(htmlTocPathTs).then(m => resolve(m.createTocXhtmlStr(this.options)))),
+      ]).then(([data1, data2, data3]) => {
+        fs.writeFileSync(path.resolve(this.uuid, './OEBPS/content.opf'), data1);
+        fs.writeFileSync(path.resolve(this.uuid, './OEBPS/toc.ncx'), data2);
+        fs.writeFileSync(path.resolve(this.uuid, './OEBPS/toc.xhtml'), data3);
+        return resolve();
+      },
+        (err) => {
+          console.error(err);
+          return reject(err);
+        }
+      );
+    });
+  }
+
+  makeCover(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      if (this.options.cover) {
+        const destPath = path.resolve(
+          this.uuid,
+          './OEBPS/cover.' + this.options._coverExtension
+        );
+        let writeStream = null;
+        //NOTE: Excluded http protocol implementation (to grab http based cover)
+        let resolvedCover = path.resolve(this.options.cover);
+        writeStream = fs.createReadStream(resolvedCover);
+        writeStream.pipe(fs.createWriteStream(destPath));
+
+
+        writeStream.on('end', () => {
+          if (this.options.verbose) {
+            console.log('[Success] cover image downloaded successfully!');
+          }
+          return resolve();
+        });
+        writeStream.on('error', (err) => {
+          console.error('Error', err);
+          return reject(err);
+        });
+      } else {
+        resolve();
+      }
+    });
+  }
+
+  getImage(options: IImage): Promise<IImage | null> {
+    if (!options.url && typeof options !== 'string') {
+      console.warn('No {Options.url} provided ..');
+      return Promise.resolve(null);
+    }
+    return new Promise((resolve, reject) => {
+      const filename = path.resolve(
+        this.uuid,
+        './OEBPS/images/' + options.id + '.' + options.extension
+      );
+      if (options.url.indexOf('file://') === 0) {
```

you should throw an error if the protocol of the URL is not `file`.
You should wrap the URL into a `new URL(url)` where `const {URL} from 'url'`
You can access the protocol on the class